reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Amb.kt

```diff
@@ -3,8 +3,13 @@ package com.badoo.reaktive.completable
 import com.badoo.reaktive.base.CompositeDisposableObserver
 import com.badoo.reaktive.utils.atomic.AtomicBoolean
 
-fun Iterable<Completable>.amb(): Completable =
+fun Collection<Completable>.amb(): Completable =
```

Comment:
Ok, I see we have different approaches in different such operators. E.g. in `concat` we are copying the `Iterable` into `List`. In `combineLatest` we have `Collection` and just iterating over it. In `amb` we have `Iterable` but we also need `isEmpty`.

So let's do the following: use `Iterable` everywhere and copy it `toList()` on susbcribe (like in `concat`). This will give us consistent API and all benefits of `Collection`. Also "consistency" with RxJava and probably safety (since collections can be potentially mutable under the hood).

Reply:
I will wait for this to be applied and will continue reviewing the PR after that. Otherwise please let me know. Thanks!

Reply:
That's a good point, but with `receiver` and `sources` it's easy to make a mistake and for example to call `forEach` on `receiver`, which can work wrong only with mutable receiver.

In my opinion there are four solutions:
- inside "subscribe" wrap everything in `with(toList())` (i prefer this one)
- write test for every such case with mutable receiver
- write two functions: `fun Iterable.amb() = toList.amb()` and `private fun List.amb()`
- leave this to reviewers

Reply:
Sorry for a delay with the response. I think the `concat` is already fine and there is no need to touch it. Affected operators iterate only once so just be careful there. Please keep in mind that collections should be converted lazily - inside `subscribe` block.

Reply:
Ok, done.